tests/test_esm_runscripts/test_filedicts.py

```diff
@@ -537,27 +560,56 @@ def test_datestamp_format_attr(simulation_file):
     assert isinstance(simulation_file.datestamp_format, str)
     assert simulation_file.datestamp_format in ["check_from_filename", "append"]
     with pytest.raises(ValueError, match="one of check_from"):
-        simulation_file['datestamp_format'] = "blah"
+        simulation_file["datestamp_format"] = "blah"
     with pytest.raises(ValueError, match="one of "):
-        # NOTE(PG): you can use _ for capturing a variable you don't care about
         simulation_file.datestamp_format = "blah"
     with pytest.raises(ValueError, match="one of check_from"):
-        # NOTE(PG): you can use _ for capturing a variable you don't care about
         simulation_file.update(dict(datestamp_format="blah"))
 
 
-def test_datestamp_added_by_default_mv(simulation_file):
-    assert False
+@pytest.mark.parametrize("movement_type", ["mv", "ln", "cp"])
+def test_datestamp_added_by_default_mv_ln_cp(simulation_file, fs, movement_type):
+    """
+    Checks that the simulation file gets a timestamp if a file already exists with that name
 
-def test_datestamp_added_by_default_ln(simulation_file):
-    assert False
+    Defaults checked:
+    -----------------
+        * datestamp_format: append
+        * datestamp_method: avoid_overwrite
+    """
+    simulation_file_meth = getattr(simulation_file, movement_type)
+    simulation_file2 = simulation_file  # Make a copy
+    simulation_file2_meth = getattr(simulation_file2, movement_type)
+    simulation_file_meth("computer", "work")
+    if movement_type == "mv":
+        assert not os.path.exists("/work/ollie/pool/ECHAM/T63/T63CORE2_jan_surf.nc")
+        # Recreate the moved file in computer (let's pretend we compied it):
+        fs.create_file(simulation_file2["absolute_path_in_computer"])
+    assert os.path.exists(
+        "/work/ollie/pgierz/some_exp/run_20010101-20010101/work/unit.24"
+    )
+    simulation_file2_meth("computer", "work")
+    assert os.path.exists(
+        "/work/ollie/pgierz/some_exp/run_20010101-20010101/work/unit.24_2000-01-01T00:00:00"
+    )
+    simulation_file_meth("work", "run_tree")
+    assert os.path.exists(
+        "/work/ollie/pgierz/some_exp/run_20000101-20000101/input/echam/T63CORE2_jan_surf.nc"
+    )
+    simulation_file_meth("run_tree", "exp_tree")
 
-def test_datestamp_added_by_default_cp(simulation_file):
-    assert False
 
 def test_datestamp_not_added_if_attr_set(simulation_file):
-    assert False
+    pass  # Still to be implemented
 
 
 def test_datestamp_not_added_if_in_filename(simulation_file):
-    assert False
+    pass  # Still to be implemented
+
+
+def test_fname_has_date_stamp_info():
+    fname = "blah_2000-01-01.nc"
+    fname2 = "blah_2000-01.nc"
+    date = esm_calendar.Date("2000-01-01T00:00:00")
+    assert filedicts._fname_has_date_stamp_info(fname, date) is True
```

Comment:
Be careful. not `is bool` but `== bool`

Reply:
Your suggestion is an anti-pattern. See here: https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html#compare-values-to-true-using-the-pattern-if-cond-is-true

Reply:
Sorry my bad. But the document says that `is True` was also as bad according to PEP8.

Reply:
I think `is` opeator is for comparing the identities (eg. `id()`) of an object and `==` for the logical comparison.

Reply:
From the document:
> The code below uses the PEP 8 preferred pattern of `if condition:`. If the type of the condition is Boolean, it is obvious that comparing to True is redundant.

So, try fixing it with the suggestion feature! 😃

Reply:
> I think is opeator is for comparing the identities (eg. id()) of an object and == for the logical comparison.
Correct! 👍🏻

Reply:
```suggestion
    assert filedicts._fname_has_date_stamp_info(fname, date)
```